moveit_ros/planning/rdf_loader/src/synchronized_string_parameter.cpp

```diff
@@ -123,19 +127,17 @@ bool SynchronizedStringParameter::waitForMessage(const rclcpp::Duration timeout)
       name_, rclcpp::QoS(1).transient_local().reliable(),
       std::bind(&SynchronizedStringParameter::stringCallback, this, std::placeholders::_1));
 
-  rclcpp::WaitSet wait_set;
-  wait_set.add_subscription(string_subscriber_);
+  auto timeout_ns =
+      std::chrono::duration_cast<std::chrono::nanoseconds>(timeout.to_chrono<std::chrono::duration<double>>());
+  auto end = std::chrono::steady_clock::now() + timeout_ns;
 
-  auto ret = wait_set.wait(timeout.to_chrono<std::chrono::duration<double>>());
-  if (ret.kind() == rclcpp::WaitResultKind::Ready)
+  while (std::chrono::steady_clock::now() < end)
   {
-    std_msgs::msg::String msg;
-    rclcpp::MessageInfo info;
-    if (string_subscriber_->take(msg, info))
+    if (queue_.pop(content_))
     {
-      content_ = msg.data;
       return true;
     }
+    std::this_thread::sleep_for(0.01 * timeout_ns);
```

Comment:
No magic numbers.  Make this a constant at the top of the file like this:

```suggestion
    // fraction of timeout used for period of polling loop 
    constexpr auto POLL_RATE = 0.01;
    ... 
    std::this_thread::sleep_for(POLL_RATE * timeout_ns);
```